docs/testingplatform/registertestframework.md

```diff
@@ -63,4 +64,4 @@ return await testApplication.RunAsync();
 ```
 
 > [!NOTE]
-> Return empty [ITestFrameworkCapabilities](capabilities.md) should not hinder the execution of the test session. The fundamental features of discovering and running tests should always be ensured. The impact should be limited to extensions that may opt out if the test framework lacks a certain feature.
+> Returning empty [ITestFrameworkCapabilities](capabilities.md) should not prevent the execution of the test session. All test frameworks should be capable of discovering and running tests. The impact should be limited to extensions that may opt out if the test framework lacks a certain feature.
```

Comment:
I am actually not fully sure what you meant here. 

> The impact should be limited to extensions that may opt out if the test framework lacks a certain feature.

Do you mean that test framework announces capabilities that are additional to discover, and run. And other extensions can use those capabilities to disable themselves based on that? E.g. if my framework can (non-conditionally) provide source code file paths (e.g. via source gen), and I have extension that relies on source code file paths, I can use the capability to enable or disable my extension.

Reply:
For instance if your adapter doesn't support the trx capability your tests will run or be discovered and simply if the trx extension is involved will be a trx extension duty decide what to do. Today for instance we print in console something like "trx will be subotimal"

Reply:
> And other extensions can use those capabilities to disable themselves based on that?

Not only disable but if implemented get the needed information to work correctly.